associacao-brasileira-de-normas-tecnicas.csl

```diff
@@ -153,120 +161,172 @@ do autor. Na regra da ABNT o sobrenome deve aparecer com todas as letras em caix
   <macro name="title">
     <choose>
-      <if type="chapter bill" match="any">
-        <text variable="title"/>
+      <if type="chapter bill webpage post-weblog article-newspaper article-magazine article-journal motion_picture broadcast paper-conference speech" match="any">
+        <text variable="title" text-case="capitalize-first"/>
       </if>
-      <else-if type="book thesis" match="any">
-        <text variable="title" font-weight="bold"/>
-      </else-if>
-      <else-if type="article-newspaper article-magazine article-journal" match="any">
-        <text variable="title"/>
+      <else-if match="any" type="entry-encyclopedia entry-dictionary">
+        <choose>
+          <if match="any" variable="author editor collection-editor translator">
+            <text variable="title" text-case="capitalize-first"/>
+          </if>
+          <else>
+            <text variable="title" text-case="uppercase"/>
+          </else>
+        </choose>
       </else-if>
       <else>
-        <text variable="title" font-weight="bold"/>
+        <text variable="title" font-weight="bold" text-case="capitalize-first"/>
       </else>
     </choose>
   </macro>
-  <!-- Titulo dos Anais-->
+  <!-- Macro de titulo do container (como p. ex. título dos Anais) -->
   <macro name="container-title">
+    <text variable="container-title" font-weight="bold"/>
+  </macro>
+  <!-- Macro com informações do container. Se houver um container-title, essa macro coloca a expressão "In:" (em fonte normal, como a ABNT pede), seguida dos autores do container, seguidos do título do container. -->
+  <macro name="container">
     <choose>
-      <if type="paper-conference" match="any">
-        <text variable="container-title" suffix=". "/>
-        <text value="Anais" font-weight="bold"/>
-        <text value="..."/>
+      <if match="any" variable="container-title">
+        <!-- In: -->
+        <text term="in" text-case="capitalize-first" suffix=": "/>
+        <group>
+          <!-- Nomes de editores -->
+          <text macro="container-contributors"/>
+          <text macro="secondary-contributors"/>
+          <text macro="container-title"/>
+        </group>
       </if>
-      <else>
-        <text variable="container-title" font-weight="bold"/>
-      </else>
     </choose>
   </macro>
-  <!--A macro 'publisher' e responsavel por mostrar o nome da editora responsavel pela publicacao-->
+  <!--A macro 'publisher' mostra lugar, editora e data. Artigos devem colocar o lugar no campo "Extra" do Zotero (campo CSL: "note"). Em palestra (speech), colocar promotor no campo "Extra" -->
   <macro name="publisher">
     <choose>
-      <if match="any" variable="publisher-place publisher">
+      <if type="paper-conference" match="none">
         <choose>
-          <if variable="publisher-place">
-            <text variable="publisher-place" suffix=": "/>
+          <if match="any" variable="publisher-place publisher issued note">
+            <choose>
```

Comment:
This whole things is better handled in two groups with delimiters rather than with suffixes. It's more robust and less complicated.